webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx

```diff
@@ -47,8 +47,8 @@ export type Props = {
     subMenu?: Menu[];
     subMenuClass?: string;
     icon?: React.ReactNode;
-    action?: (id?: string) => void;
-    filter?: (id?: string) => boolean;
+    action?: (id: string) => void;
+    filter?: (id: string) => boolean;
```

Comment:
How hard would it be to change the type definition for these plugins to make the id parameter optional to match how we were calling this before? I don't know why this component takes a `postId` to begin with, and we're still wanting to get rid of these components, but that's the one functional change I've seen in this PR so far, so I'm leery of it

Reply:
These are mainly used in the post actions menu. So an id should be sent since you are acting on a post. (So this is not a generic submenu, but that particular submenu)

It is in theory safer to send the string than string or undefined, but I agree with you it seems a functional change.

I prefer to keep this functional change because it is safer and makes more sense to the developer (what means not receiving a post id in this context?), but happy to hear other thoughts.

Reply:
That makes sense! There's also only a really small chance that this would actually cause an issue since it'd require a very specific set of circumstances to happen